internal/datastore/options/options.go

```diff
@@ -4,14 +4,26 @@ import (
 	v0 "github.com/authzed/authzed-go/proto/authzed/api/v0"
 )
 
-//go:generate go run github.com/ecordell/optgen -output zz_generated.query_options.go . QueryOptions
+//go:generate go run github.com/ecordell/optgen -output zz_generated.query_options.go . QueryOptions ReverseQueryOptions
 
 // QueryOptions are the options that can affect the results of a normal forward query.
 type QueryOptions struct {
 	Limit    *uint64
 	Usersets []*v0.ObjectAndRelation
 }
 
+// ReverseQueryOptions are the options that can affect the results of a reverse query.
+type ReverseQueryOptions struct {
+	ReverseLimit *uint64
```

Comment:
Why `ReverseLimit`? Just to make sure the options don't collide?

Reply:
Yeah, optgen would have generated two `WithLimit` methods if they were both called `Limit`.